### PR TITLE
Shop screen

### PR DESCRIPTION
**Pull request by Ada-Au**

### CODE REVIEW

File: src/shop.cpp [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    char key[2];
+    bool bought = false;
+    tryAgainInShop(key);
+    while (key[0] != '4') {
+        string choice;    // string for two digit nums
+        int pos;
+        int amount;
+        switch (key[0]) {
+        case '1':
+            renderNpc("Charon", "Sell somehing?");
```
Comment (Ada-Au) [inline src/shop.cpp]: sell "something" =_=, also can u add a choice to quit selling here? Like other than choosing item, add q to exit.

File: src/shop.cpp [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+                    case 2:
+                        renderNpc("Charon", "You want to know what to do here?");
+                        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+                        renderNpc("Charon", "Normally, you should be suffering");
+                        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+                        renderNpc("Charon", "from the afterlife torturement.");
+                        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+                        renderNpc("Charon", "But seems like you are lucky enough");
+                        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+                        renderNpc("Charon", "to still stand here.");
```
Comment (Ada-Au) [inline src/shop.cpp]: stand here still.

File: src/shop.cpp [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+                        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+                        renderNpc("Charon", "Earn money.");
+                        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+                        renderNpc("Charon", "And GIVE them to me :D");
```
Comment (Ada-Au) [inline src/shop.cpp]: can u combine the two, seems like they are pretty short

File: src/shop.cpp [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+}
+
+void talkAgain(string str, bool &isTalking, int &c, bool isFirst) {
+    string choice = " ";
+    while ( (choice[0] != '1' && choice[0] != '2' && choice[0] != '3' && choice[0] != '4' && choice[0] != '5') || choice[1]!='\0') {
+        renderNpc("Charon", str);
+        std::cout << "1 - Say Hello";
+        if (!isFirst) {cout << " (again)";}
+        std::cout << "\n" << "2 - What to do here?\n" << "3 - Why are you here?\n" 
+                << "4 - Life advice(?)\n" << "5 - (Nothing wanna ask)\n";
```
Comment (Ada-Au) [inline src/shop.cpp]: nothing to ask

File: src/shop.cpp [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+                    break;
+                }
+            }
+            break;
+        case '2':
+            while (shopItems.size() < 12)
+                generateShopItems(player, shopItems);
+            displayShopItems(shopItems);
+            std::cout << "You have " << player.coin << " G." << endl;
+            renderNpc("Charon", "What do you wanna buy?");
```
Comment (Ada-Au) [inline src/shop.cpp:223]: also add quit option back to choose what to do? (but, sell, talk etc.)

**Review by Ada-Au — COMMENTED** [review]
Please update these:D

File: src/shop.cpp (hunk 1 above)
Comment (Humen-Chau) [reply src/shop.cpp]: thx for remind, I fix the typo and add quit function now

**Pull request merged**